imap_processing/tests/conftest.py

```diff
@@ -15,6 +19,119 @@ def _set_global_config(monkeypatch, tmp_path):
     )
 
 
+@pytest.fixture(scope="session")
+def monkeypatch_session():
+    from _pytest.monkeypatch import MonkeyPatch
+
+    m = MonkeyPatch()
+    yield m
+    m.undo()
+
+
 @pytest.fixture(scope="session")
 def imap_tests_path():
     return imap_module_directory / "tests"
+
+
+# Furnishing fixtures for testing kernels
+# ---------------------------------------
+@pytest.fixture(autouse=True)
+def _autoclear_spice():
+    """Automatically clears out all SPICE remnants after every single test to
+    prevent the kernel pool from interfering with future tests. Option autouse
+    ensures this is run after every test."""
+    yield
+    spice.kclear()
+
+
+@pytest.fixture(scope="session")
+def spice_test_data_path(imap_tests_path):
+    return imap_tests_path / "spice/test_data"
+
+
+@pytest.fixture()
+def furnish_test_lsk(spice_test_data_path):
+    """Furnishes (temporarily) the testing LSK"""
+    test_lsk = spice_test_data_path / "naif0012.tls"
+    spice.furnsh(test_lsk)
+    yield test_lsk
+    spice.kclear()
+
+
+@pytest.fixture()
+def furnish_sclk(spice_test_data_path):
+    """Furnishes (temporarily) the SCLK for JPSS stored in the package data directory"""
+    test_sclk = spice_test_data_path / "imap_sclk_0000.tsc"
+    spice.furnsh(test_sclk)
+    yield test_sclk
+    spice.kclear()
+
+
+@pytest.fixture()
+def use_test_metakernel(tmpdir_factory, monkeypatch, spice_test_data_path):
```

Comment:
Do we want to make use of any of the `IMAP_DATA` directory locations for spice file paths?
https://github.com/IMAP-Science-Operations-Center/imap-data-access/blob/ff6c637533d39a8c5359f2e456e918f2495b0871/imap_data_access/file_validation.py#L354

You might be able to use that location as your tmpdir by accessing `imap_data_access.config["DATA_DIR"] / imap/spice/...` .
https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/b9abcb35e0dbc20a6afc5d80f6bf114200f804d3/imap_processing/tests/conftest.py#L10-L15

Reply:
Hmmm, maybe. I would like to add this to a discussion. I'm not sure I fully grok what you are suggesting but think that it might work better than using this template metakernel.

Reply:
Sorry, I was not very clear above. I meant to write your template metakernel to that `DATA_DIR` location (so just pointing to a different tmpdir location, not changing any logic). The `DATA_DIR` in tests is already set to a temporary directory, so it'd just avoid another temporary directory/location and possibly be closer to where we want the kernels during the mission I think.